mm2src/coins/rpc_command/tendermint/staking.rs

```diff
@@ -0,0 +1,149 @@
+        match e {
+            TendermintCoinRpcError::InvalidResponse(e)
+            | TendermintCoinRpcError::PerformError(e)
+            | TendermintCoinRpcError::RpcClientError(e) => ValidatorsRPCError::Transport(e),
+            TendermintCoinRpcError::Prost(e) | TendermintCoinRpcError::InternalError(e) => ValidatorsRPCError::InternalError(e),
+            TendermintCoinRpcError::UnexpectedAccountType { .. } => ValidatorsRPCError::InternalError(
+                "RPC client got an unexpected error 'TendermintCoinRpcError::UnexpectedAccountType', this isn't normal."
+                    .into(),
+            ),
+        }
+    }
+}
+
+pub async fn validators_rpc(
+    ctx: MmArc,
+    req: ValidatorsRPC,
+) -> Result<ValidatorsRPCResponse, MmError<ValidatorsRPCError>> {
+    fn maybe_jsonize_description(description: Option<Description>) -> Option<serde_json::Value> {
+        description.map(|d| {
+            json!({
+                "moniker": d.moniker,
+                "identity": d.identity,
+                "website": d.website,
+                "security_contact": d.security_contact,
+                "details": d.details,
+            })
+        })
+    }
+
+    fn maybe_jsonize_commission(commission: Option<Commission>) -> Option<serde_json::Value> {
+        commission.map(|c| {
+            let rates = c.commission_rates.map(|cr| {
+                json!({
+                    "rate": cr.rate,
+                    "max_rate": cr.max_rate,
+                    "max_change_rate": cr.max_change_rate
+                })
+            });
+
+            json!({
+                "commission_rates": rates,
+                "update_time": c.update_time
+            })
+        })
+    }
+
+    fn jsonize_validator(v: Validator) -> serde_json::Value {
+        json!({
+            "operator_address": v.operator_address,
+            "consensus_pubkey": v.consensus_pubkey,
+            "jailed": v.jailed,
+            "status": v.status,
+            "tokens": v.tokens,
+            "delegator_shares": v.delegator_shares,
+            "description": maybe_jsonize_description(v.description),
+            "unbonding_height": v.unbonding_height,
+            "unbonding_time": v.unbonding_time,
+            "commission": maybe_jsonize_commission(v.commission),
+            "min_self_delegation": v.min_self_delegation,
+        })
+    }
+
+    let validators = match lp_coinfind_or_err(&ctx, &req.coin).await {
+        Ok(MmCoinEnum::Tendermint(coin)) => coin.validators_list(req.filter_by_status, req.paging).await?,
+        Ok(MmCoinEnum::TendermintToken(token)) => {
+            token
+                .platform_coin
+                .validators_list(req.filter_by_status, req.paging)
+                .await?
+        },
+        Ok(_) => return MmError::err(ValidatorsRPCError::UnexpectedCoinType { ticker: req.coin }),
+        Err(_) => return MmError::err(ValidatorsRPCError::UnexpectedCoinType { ticker: req.coin }),
```

Comment:
```rust
_ => return MmError::err(ValidatorsRPCError::UnexpectedCoinType { ticker: req.coin }),
```

Reply:
One of them suppose to be `CoinNotFound`, thanks for pointing this :)